### PR TITLE
[Admin] Tax rate validation features

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

Based on #4624

### CODE REVIEW

File: src/Sylius/Behat/Context/Setup/TaxationContext.php [hunk 1]
```diff
@@ -81,27 +81,36 @@ public function __construct(
...
      * @Given the store has :taxRateName tax rate of :taxRateAmount% for :taxCategoryName within :zone zone
+     * @Given the store has :taxRateName tax rate of :taxRateAmount% for :taxCategoryName within :zone zone identified by :taxRateCode code 
      * @Given /^the store has "([^"]+)" tax rate of ([^"]+)% for "([^"]+)" for (the rest of the world)$/
      */
-    public function storeHasTaxRateWithinZone($taxRateName, $taxRateAmount, $taxCategoryName, ZoneInterface $zone)
+    public function storeHasTaxRateWithinZone($taxRateName, $taxRateAmount, $taxCategoryName, ZoneInterface $zone, $taxRateCode = null)
     {
         $taxCategory = $this->getOrCreateTaxCategory($taxCategoryName);
 
+        if(null === $taxRateCode) {
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Setup/TaxationContext.php]: Missing space after ``if``.

File: src/Sylius/Behat/Page/Admin/Crud/CreatePage.php [hunk 2]
```diff
@@ -51,7 +52,12 @@ public function create()
      */
     public function checkValidationMessageFor($element, $message)
     {
-        return $message === $this->getElement($element)->getParent()->find('css', '.pointing')->getText();
+        $foundedElement = $this->getElement($element)->getParent()->find('css', '.pointing');
+        if(null === $foundedElement) {
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/Crud/CreatePage.php]: Again.

File: src/Sylius/Behat/Page/Admin/TaxRate/CreatePage.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        'code' => '#sylius_tax_rate_code',
+        'name' => '#sylius_tax_rate_name',
+        'category' => '#sylius_tax_rate_category',
+        'zone' => '#sylius_tax_rate_zone',
+        'amount' => '#sylius_tax_rate_amount',
+        'calculator' => '#sylius_tax_rate_calculator',
+    ];
+
+    /**
+     * {@inheritDoc}
```
Comment (Zales0123) [inline src/Sylius/Behat/Page/Admin/TaxRate/CreatePage.php]: ``{@inheritdoc}`` here and below.

**Comment by pjedrzejewski** [comment]
@lchrusciel Requires rebasing.

**Comment by lchrusciel** [comment]
@pjedrzejewski done

File: src/Sylius/Behat/Context/Ui/Admin/ManagingTaxRateContext.php [hunk 4]
```diff
@@ -279,6 +292,51 @@ public function thisTaxRateShouldBeApplicableInZone(TaxRateInterface $taxRate, $
     }
 
     /**
+     * @Then I should be notified that :element has to be selected
+     */
+    public function iShouldBeNotifiedThatZoneIsRequired($element)
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingTaxRateContext.php]: ``iShouldBeNotifiedThatElementIsRequired``?

**Comment by pjedrzejewski** [comment]
Thanks Łukasz!

**Pull request merged**